src/nodedef.cpp

```diff
@@ -491,7 +498,21 @@ void ContentFeatures::serialize(std::ostream &os, u16 protocol_version) const
 	writeU32(os, damage_per_second);
 
 	// liquid
-	writeU8(os, liquid_type);
+	LiquidType liquid_type_bc = liquid_type;
+	if (protocol_version <= 39) {
+		// Since commit 7f25823, liquid drawtypes can be used even with LIQUID_NONE
+		// solution: force liquid type accordingly to accepted values
+		switch (drawtype) {
+			case NDT_LIQUID:
+				liquid_type_bc = LIQUID_SOURCE;
+				break;
+			case NDT_FLOWINGLIQUID:
+				liquid_type_bc = LIQUID_FLOWING;
+				break;
+			default: break;
+		}
```

Comment:
this is actually shorter as an `if`

Reply:
However it is less readable IMO. Changed anyway.